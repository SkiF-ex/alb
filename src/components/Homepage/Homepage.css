#section {
    position: relative;
    height: 800px;
    width: 600px;
    display: flex;
}

.homepage_title {
    height: 500px;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
}

.homepage {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    height: 100%;
    width: 100%;
    border: 1px solid black;
}

.get_stickers {
    position: absolute;
    border: 1px solid black;
    bottom: -2px;
    right: -380px;
    font-size: 25px;
    padding: 10px;
    font-weight: bold;
    background-color: transparent;
}

.homepage_name {
    position: absolute;
    bottom: 0;
    right: 20px;
    font-size: 50px;
    margin-bottom: 10px;
}

.homepage_heading {
    font-size: 100px;
    margin: 0;
}

.homepage_heading-item {
    font-size: 100px;
    margin: 0 0 0 200px;
    color: rgb(248, 105, 10);
}

.stickers_window {
    width: 1000px;
    height: 400px;
    border: 1px solid black;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    position: absolute;
    top: 25%;
    left: 25%;
    background-color: white;
    z-index: 3;
}

.stickers_block {
    display: flex;
}

.sticker_item {
    width: 150px;
    height: 250px;
    margin: 0 5px;
    display: flex;
    align-items: center;
    flex-direction: column;
}

.sticker_pack {
    width: 150px;
    height: 250px;
    margin: 0 40px;
    display: flex;
    flex-direction: column;
}

.sticker_blank {
    position: relative;
    height: 100%;
    border: 1px solid black;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
}

.sticker_button {
    margin-top: 20px;
    background-color: transparent;
    border: 1px solid black;
    padding: 10px;
    font-weight: bold;
    font-size: 20px;
}

.sticker_title {
    font-size: 20px;
    margin: -40px 0 0 0;
    font-weight: bold;
}

.sticker_subtitle {
    font-size: 20px;
    margin: 0 0 0 35px;
    font-weight: bold;
    color: rgb(248, 105, 10);
}

.sticker_set {
    position: absolute;
    bottom: -10px;
    font-weight: bold;
    font-size: 20px;
}

.stickers_header_title, .stickers_header_subtitle {
    margin: 0;
    font-size: 30px;
    font-weight: bold;
}

.stickers_header_subtitle {
    margin-bottom: 30px;
}

.gray_window {
    z-index: 2;
    background-color: rgba(61, 60, 60, 0.39);
    height: 101vh;
    width: 110%;
    overflow-x: none;
    overflow-y: none;
    position: fixed;
    margin: -10px;
    padding: 0;
}