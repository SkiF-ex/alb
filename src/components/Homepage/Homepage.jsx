import React from "react";
import { useState } from 'react';
import { Link } from "react-router-dom";
import Daily from '../Daily/Daily';
import './Homepage.css';

const Homepage = () => {
    const [dailyStickersVisibility, setDailyStickersVisibility] = useState(false);

    return (
        <>
            {dailyStickersVisibility ? (<><Daily /><div onClick={() => setDailyStickersVisibility(!dailyStickersVisibility)} className="gray_window" /></>) : null}
            <button className="button_hidden button_item" />
            <section id="section">
                <div className="homepage">
                    <div className="homepage_title">
                        <h1 className="homepage_heading">trendyol</h1>
                        <h2 className="homepage_heading-item">tech</h2>
                    </div>
                    <h3 className="homepage_name">sticker album</h3>
                </div>
                <button className="get_stickers" onClick={() => setDailyStickersVisibility(!dailyStickersVisibility)}>GET YOUR DAILY STICKERS</button>
            </section>
            <Link className="button_item next_page" to="/team">
              <button className="button_item" />
            </Link>
        </>
    )
}

export default Homepage;
