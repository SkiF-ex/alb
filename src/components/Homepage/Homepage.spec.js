import Homepage from "./Homepage";
import { act, create } from 'react-test-renderer';
import React from "react";
import {MemoryRouter} from "react-router";
import {shallow} from "enzyme";

describe('Homepage component', () => {
  it('should render component', () => {
    let component;
    act(() => {
      component = create(
        <MemoryRouter>
          <Homepage />
        </ MemoryRouter>
      );
    });
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should find .gray_window', () => {
    const component = shallow(<Homepage />);
    component.find('.get_stickers').simulate('click');

    expect(component.find('.gray_window').length).toBe(1);
  });

  it('should close getStickers window', () => {
    const component = shallow(<Homepage />);
    component.find('.get_stickers').simulate('click');
    component.find('.gray_window').simulate('click');

    expect(component.find('.gray_window').length).toBe(0);
  });
})
