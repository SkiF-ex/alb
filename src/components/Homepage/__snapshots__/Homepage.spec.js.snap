// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage component should render component 1`] = `
Array [
  <button
    className="button_hidden button_item"
  />,
  <section
    id="section"
  >
    <div
      className="homepage"
    >
      <div
        className="homepage_title"
      >
        <h1
          className="homepage_heading"
        >
          trendyol
        </h1>
        <h2
          className="homepage_heading-item"
        >
          tech
        </h2>
      </div>
      <h3
        className="homepage_name"
      >
        sticker album
      </h3>
    </div>
    <button
      className="get_stickers"
      onClick={[Function]}
    >
      GET YOUR DAILY STICKERS
    </button>
  </section>,
  <a
    className="button_item next_page"
    href="/team"
    onClick={[Function]}
  >
    <button
      className="button_item"
    />
  </a>,
]
`;
